KiedyKolos/Common/KiedyKolos.Core/Validators/YearCourse/UpdateYearCourseCommandValidator.cs

```diff
@@ -0,0 +1,49 @@
+using System.Linq;
+using FluentValidation;
+using KiedyKolos.Core.Commands;
+using KiedyKolos.Core.Interfaces;
+
+namespace KiedyKolos.Core.Validators.YearCourse
+{
+    public class UpdateYearCourseCommandValidator : AbstractValidator<UpdateYearCourseCommand>
+    {
+        private readonly IUnitOfWork _unitOfWork;
+
+        public UpdateYearCourseCommandValidator(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+
+            RuleFor(x => x).MustAsync(async (x, cancellation) => {
+                var yearCourse = await _unitOfWork.YearCourseRepository.GetAsync(x.YearCourseId);
+                return yearCourse != null;
+            }).WithMessage("Given year course does not exist");
+
+            RuleFor(x => x).MustAsync(async (x, cancellation) => {
+                var yearCourse = await _unitOfWork.YearCourseRepository.GetAsync(x.YearCourseId);
+                return yearCourse.Password == x.Password;
+            }).WithMessage("Wrong password for given year course");
```

Comment:
This one is reduntant, password will and should be only verified by auth middleware.